src/Param.php

```diff
@@ -84,9 +84,13 @@ public function setParams(array $params)
      *
      * @return $this
      */
-    public function addParam($key, $value)
+    public function addParam($key, $value, ?string $subKey = null)
```

Comment:
Nice addition!

Reply:
It simplifies existing code as it avoids to redefine `toArray` methods to handle this case

Reply:
It simplifies it a LOT!